license/src/com/untangle/app/license/LicenseManagerImpl.java

```diff
@@ -1265,4 +1290,38 @@ private String getServerParams()
             "&numDevices=" + numDevices +
             "&version=" + uvmVersion;
     }
+
+    /**
+     * This hook is called when license settings change
+     */
+    private class LicenseChangeHookCallback implements HookCallback {
+
+        /**
+         * Constructor
+         */
+        LicenseChangeHookCallback(){}
+
+        /**
+         * Get the name of callback hook
+         *
+         * @return The name
+         */
+        public String getName() {
+            return "license-mananger-license-change-hook";
+        }
+
+        /**
+         * Callback function
+         *
+         * @param args
+         *        Callback arguments
+         */
+        public void callback(Object... args) {
```

Comment:
is there a way to add some synchronization with the callback, to prevent any issues arising from multiple calls firing on the hook at once (ie: if(itsNotRunning) {sync; auto install; shutdown invalid;})

Reply:
Not needed it looks like since we're calling the callbacks with a function that adds/reads them from a queue.